pkg/workloadmeta/collectors/internal/ecs/ecs.go

```diff
@@ -28,12 +28,14 @@ const (
 )
 
 type collector struct {
-	store           workloadmeta.Store
-	metaV1          *v1.Client
-	clusterName     string
-	hasResourceTags bool
-	resourceTags    map[string]resourceTags
-	seen            map[workloadmeta.EntityID]struct{}
+	store               workloadmeta.Store
+	metaV1              v1.Client
+	metaV3or4           func(metaURI, metaVersion string) v3or4.Client
```

Comment:
Can we put directly the client for `metaV3or4` rather than a function ?

```suggestion
	metaV1              v1.Client
	metaV3or4           v3or4.Client
```